server/sonar-web/src/main/js/apps/marketplace/components/LicenseEditionSet.tsx

```diff
@@ -114,42 +117,55 @@ export default class LicenseEditionSet extends React.PureComponent<Props, State>
       this.fetchLicensePreview(license);
       this.setState({ license });
     } else {
-      this.updateLicense(license, undefined, undefined);
+      this.resetLicense();
     }
   };
 
-  handleTermsCheck = (checked: boolean) =>
-    this.setState({ acceptTerms: checked }, () =>
-      this.updateLicense(this.state.license, this.state.licenseEdition, this.state.previewStatus)
-    );
+  handleTermsCheck = (checked: boolean) => {
+    this.setLicense({ acceptTerms: checked });
+  };
 
-  updateLicense = (license: string, licenseEdition?: Edition, previewStatus?: string) => {
-    this.setState({ license, licenseEdition, loading: false, previewStatus });
+  resetLicense = (state = {}) => {
```

Comment:
When you use a default value `{}`, you lose the type-checking, i.e. you can pass basically anything to this function. Instead always type such patterns with `state: Partial<State> = {}`.

Reply:
We can't use the spread operator but at least this works :
```TS
resetLicense<K extends keyof State>(state: Pick<State, K>) {
    this.setLicense(
      Object.assign(
        {
          license: '',
          licenseEdition: undefined,
          previewStatus: undefined,
          wrongEdition: false
        },
        state
      )
    );
  }
```